### ISSUE TITLE
Torque / PBS rejects jobs with names longer than 16 characters

### ISSUE THREAD

**Issue by sweitzner** [prisms-center/pbs#11]
I am not sure how universal this problem is, but our cluster rejects jobs with names that are longer than 16 characters when submitted on the commandline with the -N directive. This is not the case if the directives are saved in a script and the script’s name is longer than 16 characters. I’m waiting to hear back from the helpdesk to see if this is due to a decision made by the system admins, or if this is due to the version of torque we are using.

Below is some sample output which shows that the the job gets rejected, and that casm does not seem to catch the error:

rundir /gpfs/scratch/sew5166/AuAg_bulk/training_data/SCEL4_4_1_1_0_0_0/2/calctype.default
id: []
Constructing a Quantum Espresso Relax object
  Relax directory: /gpfs/scratch/sew5166/AuAg_bulk/training_data/SCEL4_4_1_1_0_0_0/2/calctype.default
Quantum Espresso Relax object constructed

Preparing to submit a Quantum Espresso relaxation PBS job
  Counting atoms in the POSCAR
  Constructing a PBS job
  Submitting
Specified job names must be less than 16 characters ("SCEL4_4_1_1_0_0_0.2" is 19)!
qsub: Your job has been administratively rejected by the queueing system.
qsub: There may be a more detailed explanation prior to this notice.
Wrote /gpfs/scratch/sew5166/AuAg_bulk/training_data/SCEL4_4_1_1_0_0_0/2/calctype.default/status.json
CASM QEWrapper relaxation PBS job submission complete

**Comment by bpuchala** [comment]
Just to clarify, if there is a submit script file, and inside the file is the directive `#PBS -N <jobname>`, then `<jobname>` may be 16 characters or greater?

We could potentially add an option to write a submit script file first and then submit; or we could include an option to limit the length of the jobname.  Either should be fairly easy.

**Comment by sweitzner** [comment]
I believe if -N <jobname> or #PBS -N <jobname> is used at all, it must be less than 16 characters.

The workaround would be to save a file called <jobname>, and run “qsub <jobname>”.

I can do a quick test to confirm.

**Comment by sweitzner** [comment]
Yes, if -N <jobname> or #PBS -N <jobname> is used, it must be less than 16 characters. No restriction on qsub <jobname>.

**Comment by bpuchala** [comment]
Do you have thoughts on what jobname should be used as an alternative to the complete configuration name?  Simple truncation? Constant string such as "casm"?  Something indexed, like "casm.<index>"?

**Comment by sweitzner** [comment]
Is there a type of linear indexing scheme already being used in casm for configurations? such as SCEL1_0_0_.../0 = 0, SCEL1_0_0_.../1 = 1, SCEL2_1_0_0_../0 = 2, etc.?

I suppose the original naming system made it easy for people to see which jobs are still running. Maybe this would be a good alternative? The only downside is that the indices would have to be regenerated for different selections or augmented sets of configurations.

**Comment by sweitzner** [comment]
FYI, from my helpdesk:

"I believe the length restriction on the job name was put in place to deal with non-ASCII characters, which were causing issues with job submission on our systems. With this name length restriction in place, we can detect and reject jobs containing non-ASCII characters due to how they are converted to ASCII characters.

This restriction is put in place through a job submission filter and therefore is not a feature of the job scheduler system. Do you use the python module 'pbs' to automate the submission of your batch CASM jobs (https://github.com/prisms-center/pbs)? If not, what do you use? If the code is open source, we may be able to alter it to write an intermediate job script file. "

**Comment by bpuchala** [comment]
Ah, I didn't realize that if ``#PBS -N <jobname>`` is not specified that the script name is used because I've always just specified it with ``-N``.  So we can write the submission script with the name ``SCEL4_4_1_1_0_0_0.2`` and submit that.  That is easy enough.

**Comment by sweitzner** [comment]
Yes, I think that should work too.

**Comment by bpuchala** [comment]
I've made changes that seem to work for me on the '2.X_file_submit' branch.  Could you test it on your system?

To make 'pbs' submit via file instead of command line just add ``"write_submit_script":true`` in your ``config.json`` file.

**Comment by sweitzner** [comment]
This works on my cluster. I get a different error that's due to the QE wrapper which I'll bring up with Sanjeev, but job submission works now. Thanks.

In the future, will adding ```"write_submit_script":true``` to the ```config.json``` just be a manual step?

**Comment by bpuchala** [comment]
I hadn't planned on anything more. Since the submit script is stored in the jobs database I think keeping the command line submission the default makes sense. I'm certainly open to ideas if you have changes in mind.

**Comment by sweitzner** [comment]
No, that makes sense. I was just curious if things were going to change since I need to help others in my group install CASM and the pbs module at some point.
